### PR TITLE
per device adblock

### PR DESCRIPTION
**Pull request by wzjjack**

### CODE REVIEW

File: extension/dnsmasq/dnsmasq.js [hunk 1]
```diff
@@ -692,9 +694,9 @@ module.exports = class DNSMASQ {
       if (type === "family") {
         targetIP = BLUE_HOLE_IP
       }
-
+      const tag = this.level[type] == "system" ? "" : "$ad_block";
```
Comment (jasonlyc) [inline extension/dnsmasq/dnsmasq.js]: Do not hardcode ad_block here. You can use value of 'type' as tag, here type is 'adblock'.

File: sensor/AdblockPlugin.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    async run() {
+        this.systemSwitch = false;
+        this.adminSystemSwitch = false;
+        this.enabledMacAddresses = [];
+        extensionManager.registerExtension("adblock", this, {
+            applyPolicy: this.applyPolicy,
+            start: this.start,
+            stop: this.stop
+        });
+        const isPolicyEnabled = await spt.isPolicyEnabled('adblock');
```
Comment (MelvinTo) [inline sensor/AdblockPlugin.js]: this should be one-time only. After this upgrade, if adblock policy:system flag is on and adblock feature flag is off, the feature flag should be kept as off

File: sensor/FamilyProtectPlugin.js [hunk 3]
```diff
@@ -49,6 +51,10 @@ class FamilyProtectPlugin extends Sensor {
             start: this.start,
             stop: this.stop
         });
+        const isPolicyEnabled = await spt.isPolicyEnabled('family');
```
Comment (MelvinTo) [inline sensor/FamilyProtectPlugin.js]: this should be one-time only. After this upgrade, if adblock policy:system flag is on and adblock feature flag is off, the feature flag should be kept as off

**Comment by MelvinTo** [comment]
a general comment is old/new family feature should work with old/new app. (fully compatible)

old family <-> old app (yes)
old family <-> new app (can still manage family with policy:system toggle
new family <-> old app (can still manage family with policy:system toggle, feature flag is on)
new family <-> new app (can manage family with feature flag + all devices + specified devices)

this is also applied to adblock

**Pull request merged**